py/server/deephaven/table.py

```diff
@@ -2679,13 +2684,18 @@ def transform(self, func: Callable[[Table], Table]) -> PartitionedTable:
         """
         try:
             j_operator = j_unary_operator(func, dtypes.from_jtype(Table.j_object_type.jclass))
-            with auto_locking_ctx(self):
-                j_pt = self.j_partitioned_table.transform(j_operator)
+            dependencies = to_sequence(dependencies, wrapped=True)
+            j_dependencies = [d.j_table for d in dependencies if isinstance(d, Table)]
+            j_dependencies.extend([d.table.j_table for d in dependencies if isinstance(d, PartitionedTable)])
```

Comment:
There should be a size assert because this will happily run if non Table / PartitionedTable garbage is included.

Reply:
It might be an overkill, I think it is safe to assume that users who need to use this parameter knows what they are doing.

Reply:
Will Python's type system (hah) allow that?